src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx

```diff
@@ -90,17 +90,39 @@ const RegisterForm: React.FunctionComponent<PropsType> = ({
   const onSubmit = async (data: RegisterFormData) => {
     try {
       setShowLoginModal(false);
-      if (venue.requiresTicketCode) await axios.get(codeCheckUrl(data.code));
-      if (venue.requiresEmailVerification)
-        await axios.get(codeCheckUrl(data.email));
-
       const auth = await signUp(data);
-      if (
-        auth.user &&
-        (venue.requiresTicketCode || venue.requiresEmailVerification)
-      ) {
+
+      if (venue.access?.includes(VenueAccessMode.Codes)) {
+        const result = await checkAccess({
+          venueId: venue.id,
+          code: data.code,
+        });
+        console.log("result.data", result.data);
+        console.log(
+          "saving token to",
+          accessTokenKey(venue.id),
+          ": ",
+          result.data.token
+        );
+        setLocalStorageToken(venue.id, result.data.token);
+      }
+      if (venue.access?.includes(VenueAccessMode.Emails)) {
+        const result = await checkAccess({
+          venueId: venue.id,
+          email: data.email,
+        });
+        console.log("result.data", result.data);
+        console.log(
+          "saving token to",
+          accessTokenKey(venue.id),
+          ": ",
+          result.data.token
+        );
+        setLocalStorageToken(venue.id, result.data.token);
+      }
```

Comment:
Seemingly the only difference between the code used in these 2 sections is the object of params passed to `checkAccess`? Can we refactor the common parts out of this to simplify it right down?

Reply:
There are also very similar blocks of code in `src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx` that can probably similarly be refactored to share the common parts.